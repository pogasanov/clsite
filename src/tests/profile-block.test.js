import {shallowMount} from '@vue/test-utils'
import profileBlock from '../js/components/commons/profile-block.vue'

describe('profile-block', () => {
    it('is a Vue instance', () => {
        const wrapper = shallowMount(profileBlock);
        expect(wrapper.isVueInstance()).toBeTruthy()
    });

    it('renders correctly', () => {
        const wrapper = shallowMount(profileBlock, {
            propsData: {
                title: 'Tested title',
                section_id: 'tested_id'
            },
            slots: {
                default: '<div>Tested slot</div>'
            }
        });
        expect(wrapper.element).toMatchSnapshot()
    });

    it('set title', () => {
        const EXPECTED_TITLE = 'Tested title';

        const wrapper = shallowMount(profileBlock, {
            propsData: {
                title: EXPECTED_TITLE
            }
        });
        expect(wrapper.props().title).toBe(EXPECTED_TITLE);
        expect(wrapper.find('h2').text()).toBe(EXPECTED_TITLE)
    });

    it('set section id', () => {
        const EXPECTED_ID = 'tested_id';

        const wrapper = shallowMount(profileBlock, {
            propsData: {
                section_id: EXPECTED_ID
            }
        });
        expect(wrapper.props().section_id).toBe(EXPECTED_ID);
        expect(wrapper.find('section').attributes().id).toBe(EXPECTED_ID)
    })
});
