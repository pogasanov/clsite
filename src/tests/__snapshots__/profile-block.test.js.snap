// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile-block renders correctly 1`] = `
<section
  class="profile-block"
  id="tested_id"
>
  <header
    class="profile-block__header"
  >
    <h2
      class="profile-block__title"
    >
      Tested title
    </h2>
     
    <a
      class="btn btn--small"
      href="#0"
    >
      Edit
    </a>
  </header>
   
  <div
    class="profile-block__main"
  >
    <div>
      Tested slot
    </div>
  </div>
</section>
`;
